### ISSUE TITLE
Deletion of a Shoot Namespace on Seed stuck in Terminating

### ISSUE THREAD

**Issue by bergerx** [gardener/gardener#238]
After I deleted a Project Namespace in Garden cluster and got the Shoot stuck during deletion (#237),  I manually removed the resources created on the cloud provider and wanted to remove the Shoot Namespace on the Seed. So I issued a `kubectl delete ns ...` command but I saw the Namespace stuck in deletion.

The machine-controller-manager in the same namespace is deleted before the CRDs are being handled by it, and CRDs with finalizers blocked the deletion of the Namespace. I had to go through all resources in the Namespace and manually remove the finalizers to unblock Namespace termination.

I think such scenarios could happen in production and I feel like we need to define this scenario for operators for how to handle it.

One solution could be machine-controller-manager to realise the delete is issued as a Result of Namespace termination and immediately remove its own finalizers from all resources, or also handle the deletion of machines it created. I'm not sure if there is really a viable solutions here.

**Comment by rfranzke** [comment]
We probably won't do anything here soon. Nobody should manually delete stuff in the seed, and administrators/operators that do that anyways should know what they are doing. Maybe we can implement it when we have the gardenlet, let's see.
